src/store/thunks/users.js

```diff
@@ -1,41 +1,66 @@
 import { userSignedIn, userLogout } from "../actions/users";
 
 import { googlePopupSignInMethod, db } from "database";
+import { useSelector } from "react-redux";
 
+//CREATE NEW USER
 export const signInWithGoogle = () => (dispatch) => {
   console.log("Sign in with google");
   googlePopupSignInMethod().then((res) => {
     if (res.additionalUserInfo.isNewUser) {
       const uniqueId = res.user.uid;
-      db.collection("users").doc(uniqueId).set({
-        email: res.user.email,
-        username: "",
-        posts: [],
-        full_name: res.user.displayName,
-        avatar_link: "",
-      });
+      db.collection("users")
+        .doc(uniqueId)
+        .set({
+          email: res.user.email,
+          username: "",
+          posts: [],
+          full_name: res.user.displayName,
+          avatar_link: "",
+        })
+        .then(() => {
+          handleReturningUserSignIn(res).then((user) => {
+            dispatch(userSignedIn(user));
+          });
+        });
+      console.log("added");
     } else {
+      //console.log("returning user");
       // Call a thunk here with payload of whatever the user in state needs to have
       handleReturningUserSignIn(res).then((user) => {
         dispatch(userSignedIn(user));
       });
     }
   });
 };
-
+//LOGOUT
 export const logout = () => (dispatch) => {
   dispatch(userLogout());
 };
 
+export const username = (info) => (dispatch) => {
+  console.log("thunk", info);
+  db.collection("users")
+    .doc(info.id)
+    .set({ username: info.username }, { merge: true })
+    .then(() => {
+      db.collection("users")
+        .doc(info.id)
+        .get()
+        .then((res) => {
```

Comment:
We should get into the habit of adding .catch() blocks to catch failures. For now we can console.error them or something but always a good idea to get into